dev/local-environment-dynamic/setup.sh

```diff
@@ -0,0 +1,705 @@
+        else
+            read -p "Manually configure Postgres Password? (Will generate otherwise) (Y/N): " manual_configure
+            if [[ $manual_configure =~ ^[Yy] ]]; then
+                read -sp "Enter PostgreSQL postgres user password: " db_password
+                echo
+            else
+                db_password=$(LC_ALL=C tr -dc 'a-zA-Z0-9' < /dev/urandom | head -c 20)
+                echo -e "Generated PostgreSQL postgres user password: $db_password \n"
+            fi
+        fi
+    fi
+
+    if [ "$mode" == "non-interactive" ]; then
+        db_port=5432
+    else
+        read -p "Do you want to set a non-default port for Postgres? (Will default to 5432) (Y/N): " set_db_port
+        if [[ $set_db_port =~ ^[Yy] ]]; then
+            db_port=$(validate_port "Enter port for Postgres: ")
+        else
+            db_port=5432
+        fi
+    fi
+
+    db_host=postgres
+}
+
+configure_env() {
+    local mode=$1
+
+    if [ "$mode" == "interactive" ]; then
+        echo "===== ENV FILE CREATION ============"
+        echo "Creating new .env file with environment configuration..."
+    fi
+
+    if [ "$mode" == "non-interactive" ]; then
+        cat <<EOF >.env
+POSTGRES_PORT=5432
+POSTGRES_PASSWORD=$db_password
+OGMIOS_PORT=1337
+CPU_PARTNER_CHAINS_NODE=0.000
+MEM_PARTNER_CHAINS_NODE=1000G
+CPU_CARDANO=0.000
+MEM_CARDANO=1000G
+CPU_POSTGRES=0.000
+MEM_POSTGRES=1000G
+CPU_DBSYNC=0.000
+MEM_DBSYNC=1000G
+CPU_OGMIOS=0.000
+MEM_OGMIOS=1000G
+EOF
+    else
+        cat <<EOF >.env
+POSTGRES_PORT=$db_port
+POSTGRES_PASSWORD=$db_password
+OGMIOS_PORT=$ogmios_port
+CPU_PARTNER_CHAINS_NODE=$CPU_PARTNER_CHAINS_NODE
+MEM_PARTNER_CHAINS_NODE=$MEM_PARTNER_CHAINS_NODE
+CPU_CARDANO=$cpu_cardano
+MEM_CARDANO=$mem_cardano
+CPU_POSTGRES=$cpu_postgres
+MEM_POSTGRES=$mem_postgres
+CPU_DBSYNC=$cpu_dbsync
+MEM_DBSYNC=$mem_dbsync
+CPU_OGMIOS=$cpu_ogmios
+MEM_OGMIOS=$mem_ogmios
+EOF
+    fi
+
+    cat <<EOF >>.env
+CARDANO_IMAGE=$CARDANO_IMAGE
+DBSYNC_IMAGE=$DBSYNC_IMAGE
+OGMIOS_IMAGE=$OGMIOS_IMAGE
+POSTGRES_IMAGE=$POSTGRES_IMAGE
+PARTNER_CHAINS_NODE_IMAGE=${node_image:-$PARTNER_CHAINS_NODE_IMAGE}
+NUM_PERMISSIONED_NODES_TO_PROCESS=$NUM_PERMISSIONED_NODES_TO_PROCESS
+NUM_REGISTERED_NODES_TO_PROCESS=$NUM_REGISTERED_NODES_TO_PROCESS
+EOF
+
+    if [ "$mode" == "interactive" ]; then
+        echo -e ".env file created successfully.\n"
+    fi
+}
+
+choose_deployment_option() {
+  echo "===== CUSTOM STACK MODIFICATIONS ========"
+  read -p "Make custom modification to the stack? (Y/N): " modify_stack
+  if [[ $modify_stack =~ ^[Yy]$ ]]; then
+    echo "Choose your deployment option:"
+    echo "1) Include only Cardano testnet"
+    echo "2) Include Cardano testnet with Ogmios"
+    echo "3) Include Cardano testnet, Ogmios, DB-Sync and Postgres"
+    echo "4) Deploy a single Partner Chains node with network_mode: "host" for external connections (adjust partner-chains-external-node.txt before running this script)"
```

Comment:
I've used option 4 and it started with 20 node. There was no partner-chains-external-node.txt to adjust anywhere. In local-environment this file is in "modules" directory.

If this is not supported, then please clean up